test/tests/pbs_Rrecord_resources_used.py

```diff
@@ -0,0 +1,616 @@
+        m = self.server.accounting_match(
+        msg='.*R;' + re.escape(jid3s1) + '.*resources_used.*run_count=1', id=jid3s1, regexp=True)
+
+    def test_t3(self):
+        """
+        Scenario: One non-rerunnable job. The node on which the job was running goes down and node_fail_requeue time-out is hit. 
+        Expected outcome: Server should record last known resource usage in the 'R' record only for rerunnable jobs.
+        """
+        a = {ATTR_JobHistoryEnable: 1}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+        jid1,jid2,jid3s1 = self.common(True,'r')
+
+        m = self.server.accounting_match(
+        msg='.*R;' + jid1 + '.*resources_used.*run_count=1', id=jid1, regexp=True)
+        self.assertNotEqual(m, None)
+        m = self.server.accounting_match(
+        msg='.*R;' + jid2 + '.*resources_used.*run_count=1', id=jid2, regexp=True)
+        self.assertEqual(m, None)
+        # Use re.replace() as accounting_match cannot match subjobs []
+        m = self.server.accounting_match(
+        msg='.*R;' + re.escape(jid3s1) + '.*resources_used.*run_count=1', id=jid3s1, regexp=True)
+        self.assertNotEqual(m, None)
+
+    def test_t4(self):
+        """
+        Scenario: Mom restarted without '-r' option and requeued using qrerun.
+        Expected outcome: Server should record last known resource usage in the 'R' record for both.
+        """
+        jid1,jid2,jid3s1 = self.common(False,'s')
+
+        m = self.server.accounting_match(
+        msg='.*R;' + jid1 + '.*resources_used.*run_count=1', id=jid1, regexp=True)
+        self.assertNotEqual(m, None)
+        m = self.server.accounting_match(
+        msg='.*R;' + jid2 + '.*resources_used.*run_count=1', id=jid2, regexp=True)
+        self.assertNotEqual(m, None)
+        # Use re.replace() as accounting_match cannot match subjobs []
+        m = self.server.accounting_match(
+        msg='.*R;' + re.escape(jid3s1) + '.*resources_used.*run_count=1', id=jid3s1, regexp=True)
+        self.assertNotEqual(m, None)
+        """
+        Verify that the jobs are in 'Q' state.
+        """
+        self.server.expect(JOB, {ATTR_state: 'Q'}, jid1, offset=5)
+        self.server.expect(JOB, {ATTR_state: 'Q'}, jid2)
+        self.server.expect(JOB, {ATTR_state: 'Q'}, jid3s1)
+        """
+        qrun the jobs and wait for them to start running.
+        """
+        self.server.runjob(jobid=jid1)
+        self.server.runjob(jobid=jid2)
+        self.server.runjob(jobid=jid3s1)
+
+        self.server.expect(JOB, {ATTR_state: 'R'}, jid1, offset=5)
+        self.server.expect(JOB, {ATTR_state: 'R'}, jid2)
+        self.server.expect(JOB, {ATTR_state: 'R'}, jid3s1)
+        """
+        qrerun the jobs and wait for them to start running.
+        """
+        self.server.rerunjob(jobid=jid1)
+        self.server.rerunjob(jobid=jid2)
+        self.server.rerunjob(jobid=jid3s1)
+
+        self.server.expect(JOB, {ATTR_state: 'R'}, jid1, offset=5)
```

Comment:
OK, I get it now.  If you're just interested in the accounting_match(), maybe remove the expect for job_state=R?  I don't see that as something you need to care about in the test.  I shouldn't matter, but it may get in the way of what you are really trying to test for.  Just a suggestion.